File: 프로그래머스/2/152996. 시소 짝꿍/시소 짝꿍.java
import java.util.*;

class Solution {
    public long solution(int[] weights) {
        long answer = 0;
        
        Arrays.sort(weights);
        HashMap<Double, Integer> map = new HashMap<>();
        
        for(int i=0; i<weights.length; i++){
            double arr[] = {weights[i]*1.0, weights[i]*0.5, (weights[i]*2.0)/3.0, weights[i]*0.75};
            
            for(double num : arr){
               if(map.containsKey(num)) answer += map.get(num);
            }
            
            if(map.containsKey(weights[i]*1.0)){
                map.put(weights[i]*1.0, map.get(weights[i]*1.0) + 1);
            }else{
                map.put(weights[i]*1.0, 1);
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/2/152996. 시소 짝꿍/시소 짝꿍.js
function solution(weights) {
    var answer = 0;
    
    const map = new Map();
    
    weights.sort((a,b) => a - b);
    
    for(let w of weights){
        var a = w*1.0;
        var b = w*2.0/3.0;
        var c = w*1.0/2.0;
        var d = w*3.0/4.0;
        
        if(map.has(a)) answer += map.get(a);
        if(map.has(b)) answer += map.get(b);
        if(map.has(c)) answer += map.get(c);
        if(map.has(d)) answer += map.get(d);
        
        map.set(w*1.0, (map.get(w*1.0) ?? 0) + 1);
    }
    
    return answer;
}